File: Backend/src/main/java/com/luckyseven/backend/domain/settlements/dao/SettlementRepository.java
package com.luckyseven.backend.domain.settlements.dao;

import com.luckyseven.backend.domain.settlements.entity.Settlement;
import java.util.Optional;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

@Repository
public interface SettlementRepository extends JpaRepository<Settlement, Long>,
    JpaSpecificationExecutor<Settlement> {

  @EntityGraph(attributePaths = {"settler", "payer"})
  Optional<Settlement> findWithSettlerAndPayerById(Long id);


}

File: Backend/src/main/java/com/luckyseven/backend/domain/settlements/dao/SettlementRepository.kt
package com.luckyseven.backend.domain.settlements.dao

import com.luckyseven.backend.domain.settlements.entity.Settlement
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.domain.Specification
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository

@Repository
interface SettlementRepository : JpaRepository<Settlement, Long>,
    JpaSpecificationExecutor<Settlement?> {

    @EntityGraph(attributePaths = ["settler", "payer"])
    fun findWithSettlerAndPayerById(id: Long): Settlement?

    fun findAll(spec: Specification<Settlement>, page: Pageable): Page<Settlement>
}
